lib/pg.ddl.js

```diff
@@ -0,0 +1,382 @@
+  const table =
+    `\nCREATE TABLE ${from}${name} (` +
+    `\n  ${from} bigint NOT NULL,\n  ${to} bigint NOT NULL\n);`;
+
+  const lines = [table];
+
+  console.log(from, to);
+  if (requiresForeignKey(categories, link, from)) {
+    lines.push(generateLink(name, from, from, 'Master', { category: to }));
+    lines.push(generateLink(name, to, to, 'Master', { category: from }));
+  }
+
+  lines.push(
+    `ALTER TABLE ${name} ADD CONSTRAINT pk${name}${from}${to} ` +
+      `PRIMARY KEY (${from}, ${to});`
+  );
+  return lines.join('\n');
+};
+
+const generateLinks = (links, categories) => links
+  .sort(({ link }) => link.constructor.name === 'Many')
+  .map(({ from, to, name, link }) => {
+    const type = link.constructor.name;
+    return type === 'Many' ?
+      generateManyToMany(from, to, name, link, categories) :
+      generateLink(from, to, name, type, link);
+  })
+  .join('\n');
+
+const generateIndexes = (indexes, table) => indexes
+  .map(({ name, property: index }) => {
+    const type = index.constructor.name;
+    const prop = type === 'Index' ? index.fields.join(', ') : name;
+
+    return `CREATE INDEX idx${table}${name} on ${table} (${prop});`;
+  })
+  .join('\n');
+
+const generateUnique = (unique, table) => unique
+  .map(({ name, property: index }) => {
+    const type = index.constructor.name;
+    const prop = type === 'Unique' ? index.fields.join(', ') : name;
+
+    return (
+      `ALTER TABLE ${table} ADD CONSTRAINT ak${table}${name} UNIQUE ` +
+      `(${prop});`
+    );
+  })
+  .join('\n');
+
+const generateId = (type, length) => pad('Id', length) +
+  (type === 'Global' ? ' bigint' : ' bigserial');
+
+const generateTable = (
+  name,
+  category,
+  type,
+  types,
+  unresolvedLinks,
+  existingTables,
+  categories
+) => {
+  if (!isValidIdentifier(name)) {
+    throw new Error(
+      `Cannot create table ${name} because it is not a valid identifier`
+    );
+  }
+  if (isReserved(name)) {
+    throw new Error(
+      `Cannot create table ${name} because it is a reserved keyword`
+    );
+  }
+
+  const {
+    indexes,
+    unique,
+    links,
+    properties,
+  } = categorizeEntries(category, name, categories);
+
+  const maxPropLength = getMaxPropLength(properties);
+  const resolvableLinks = unresolvedLinks.get(name) || [];
+  const unresolved = links.filter(({ name: propName, property: link }) => {
+    if (link.category === name || existingTables.includes(link.category)) {
+      resolvableLinks.push({
+        from: name,
+        to: link.category,
+        name: propName,
+        link,
+      });
+      return false;
+    }
+    return true;
+  });
+
+  const props = generateProperties(name, properties, types, maxPropLength);
+  props.unshift(generateId(type, maxPropLength));
+
+  const primaryKey =
+    `\nALTER TABLE ${name} ADD CONSTRAINT pk${name}Id PRIMARY KEY (Id);\n`;
+
+  let linksSQL = generateLinks(resolvableLinks, categories);
+  if (linksSQL) {
+    linksSQL = '\n' + linksSQL + '\n';
```

Comment:
Just a thought: it probably makes sense to also use template literals here and twice below, since you are using them extensively in this function when constructing strings.